docs/how-to-guides/data-basics/3-building-an-edit-form.md

```diff
@@ -0,0 +1,500 @@
+
+![](./media/edit-form/form-scaffold.png)
+
+Great! We now have a basic user interface to work with.
+
+### Step 3: Populate the form with page details
+
+We want the `EditPageForm` to display the title of the currently edited page. You may have noticed that it doesn't receive a `page` prop, only `pageId`. That's okay. Gutenberg Data allows us to easily access entity records from any component.
+
+In this case, we need to use the [`getEntityRecord`](/docs/reference-guides/data/data-core/#getentityrecord) selector. The list of records is already available thanks to the `getEntityRecords` call in `MyFirstApp`, so there won't even be any additional HTTP requests involved – we'll get the cached record right away.
+
+Here's how you can try it in your browser's dev tools:
+
+```js
+wp.data.select( 'core' ).getEntityRecord( 'postType', 'page', 9 );  // Replace 9 with an actual page ID
+```
+
+Let's update `EditPageForm` accordingly:
+
+```js
+export function EditPageForm( { pageId, onCancel, onSaveFinished } ) {
+	const page = useSelect(
+		select => select( coreDataStore ).getEntityRecord( 'postType', 'page', pageId ),
+		[pageId]
+	);
+	return (
+		<div className="my-gutenberg-form">
+			<TextControl
+				label='Page title:'
+				value={ page.title }
+			/>
+			{ /* ... */ }
+		</div>
+	);
+}
+```
+
+Now it should look like that:
+
+![](./media/edit-form/form-populated.png)
+
+### Step 4: Making the Page title field editable
+
+There's one problem with our _Page title_ field: you can't edit it. It receives a fixed `value` but doesn't update it when typing. We need an `onChange` handler.
+
+
+You may have seen a pattern similar to this one in other React apps:
+
+```js
+export function VanillaReactForm({ initialTitle }) {
+	const [title, setTitle] = useState( initialTitle );
+	return (
+		<TextControl
+			value={ title }
+			onChange={ setTitle }
+		/>
+	);
+}
+```
+
+Updating entity records in Gutenberg Data is similar. We don't use `useState`, though. The counterpart of `setTitle` is an `editEntityRecord` action which stores the updates in the Redux state. Here's how you can try it out in your browser's dev tools:
+
+```js
+// Get a real page ID first
+const pageId = wp.data.select( 'core' ).getEntityRecords( 'postType', 'page' )[0].id;
+
+// Uupdate the title
+wp.data.dispatch( 'core' ).editEntityRecord( 'postType', 'page', pageId, { title: 'updated title' } );
+```
+
+At this point, you may ask _How is `editEntityRecord` this better than `useState`?_ It offers a few features you wouldn't otherwise get. We can save the changes as easily as we retrieve the data and ensure that all caches will be correctly updated. In addition, the changes applied via `editEntityRecord` are easily undo-able via the `undo` and `redo` actions. Finally,  Because the changes live in the Redux state, other components can access them. For example, we could make the `PagesList` display the currently edited title.
```

Comment:
```suggestion
At this point, you may ask _how is `editEntityRecord` better than `useState`? The answer is that it offers a few features you wouldn't otherwise get. 

Firstly, we can save the changes as easily as we retrieve the data and ensure that all caches will be correctly updated. 

Secondly, the changes applied via `editEntityRecord` are easily undo-able via the `undo` and `redo` actions. 

Lastly, because the changes live in the _Redux_ state, they are "global" and can be accessed by other components. For example, we could make the `PagesList` display the currently edited title.
```

Tweaked this. If you reject the formatting changes (no problem!) please note the grammar fixes towards the beginning.